src/common/useTranslations.tsx

```diff
@@ -0,0 +1,96 @@
+import { Fragment } from "react";
+import { useTemplateData } from "src/common/useTemplateData";
+
+/**
+ * A translation function translates text using the JSON file in i18n/locale/ for the current locale.
+ * @param text The text to translate
+ * @param replacements A dictionary of replacements to make in the translated text.
+ *   The key is the placeholder to replace, and the value is the replacement.
+ *   The keys are expected to be in braces in the original text.
+ *   For example, for text "Hello {NAME}", the key should be "NAME".
+ *   The replacement value may be a string or a JSX element.
+ * @returns The translated text. If any JSX is inserted, it returns an array of JSX elements.
+ */
+export type TranslationFunction = {
+  (text: string, replacements?: Record<string, string | number>): string;
+  (text: string, replacements: Record<string, string | number | JSX.Element>):
+    | string
+    | JSX.Element[];
+};
+
+function replaceWithStringOrJSX(
+  text: string,
+  findStr: string,
+  replacement: string | JSX.Element
+): (string | JSX.Element)[] {
+  const replacedTextArr = text
+    // Split at each match
+    .split(findStr)
+    // Insert replacement between matches
+    .map((substring, i) => [i == 0 ? "" : replacement, substring])
+    .flat()
+    // Combine consecutive strings
+    .reduce((arr: (string | JSX.Element)[], strOrJSX) => {
+      if (
+        arr.length == 0 ||
+        typeof strOrJSX != "string" ||
+        typeof arr[arr.length - 1] != "string"
+      ) {
+        arr.push(strOrJSX);
+      } else {
+        arr[arr.length - 1] += strOrJSX;
+      }
+
+      return arr;
+    }, []);
+
+  return replacedTextArr;
+}
+
+function useTranslations(): TranslationFunction {
+  const { translations } = useTemplateData();
+
+  // Overloads to match the TranslationFunction type definition
+  function translate(
```

Comment:
potentially a silly question, but I thought the i18next library provided functionality like this for us?

Reply:
also piggybacking on this, do we really want to write and maintain an in-house translation library instead of using a third party library like i18next? We would probably miss a lot of functionality - just off the top of my head we probably want 
1. scanning the repo for strings to automatically generate the translation file
2. handling plural string cases
3. Switching languages as a react state update and not having to reload the page/redirect to a different page

Reply:
@bhainesva @Ceseura That sounds perfectly reasonable to me. I wrote this function because I was unfamiliar with the libraries being used in the unfinished PR and we needed something to work for Chaumet right away. I probably will not have any time to look into it since I don't have any dedicated R&D time, so are either of you able to provide examples of how to use i18next to do what this translation function is doing?